From 3fd2dcc3da6e7d693e2453945f6a3ca5826cd7b0 Mon Sep 17 00:00:00 2001
From: Nikolai Kondrashov <Nikolai.Kondrashov@redhat.com>
Date: Thu, 29 Jan 2015 14:01:49 +0200
Subject: [PATCH 1/1] Check cf_item_parse return value

Check cf_item_parse return value in dhcp_socket_parse to match all other
invocations.

This fixes the following Coverity errors:

Error: CHECKED_RETURN (CWE-252):
freeradius-server-2.2.6/src/main/dhcpd.c:618: check_return: Calling "cf_item_parse" without checking return value (as is done elsewhere 12 out of 14 times).

Error: CHECKED_RETURN (CWE-252):
freeradius-server-2.2.6/src/main/dhcpd.c:624: check_return: Calling "cf_item_parse" without checking return value (as is done elsewhere 12 out of 14 times).
---
 src/main/dhcpd.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/main/dhcpd.c b/src/main/dhcpd.c
index 350c647..ce7b6be 100644
--- a/src/main/dhcpd.c
+++ b/src/main/dhcpd.c
@@ -615,14 +615,16 @@ static int dhcp_socket_parse(CONF_SECTION *cs, rad_listen_t *this)
 	sock->suppress_responses = FALSE;
 	cp = cf_pair_find(cs, "suppress_responses");
 	if (cp) {
-		cf_item_parse(cs, "suppress_responses", PW_TYPE_BOOLEAN,
-			      &sock->suppress_responses, NULL);
+		rcode = cf_item_parse(cs, "suppress_responses", PW_TYPE_BOOLEAN,
+					&sock->suppress_responses, NULL);
+		if (rcode < 0) return -1;
 	}
 	
 	cp = cf_pair_find(cs, "src_interface");
 	if (cp) {
-		cf_item_parse(cs, "src_interface", PW_TYPE_STRING_PTR,
-			      &sock->src_interface, NULL);
+		rcode = cf_item_parse(cs, "src_interface", PW_TYPE_STRING_PTR,
+					&sock->src_interface, NULL);
+		if (rcode < 0) return -1;
 	} else {
                 sock->src_interface = sock->lsock.interface;
         }
-- 
2.1.4

